src/sysext/sysext.c

```diff
@@ -1018,7 +979,10 @@ static int run(int argc, char *argv[]) {
         if (r <= 0)
                 return r;
 
-        r = parse_env();
+        /* For debugging purposes it might make sense to do this for other hierarchies than /usr/ and
+         * /opt/, but let's make that a hacker/debugging feature, i.e. env var instead of cmdline
+         * switch. */
+        r = getenv_path_list("SYSTEMD_SYSEXT_HIERARCHIES", &arg_hierarchies);
         if (r < 0)
                 return r;
```

Comment:
hmm, so the logging got lost here now: we might fail here, but the user will not see any explanation about it — unless the user turns on debug logging...

i mean, the usual rule of "logging" vs. "non-logging" function is that if the function is "non-logging" that it's then the responsibility of the caller to log instead. but here you now don't log at all...

one option out would be to pass a log level to getenv_path_list() and use log_full() and pass the level. that works OK for smaller functions where only one log level makes sense. or call it getenv_path_list_and_warn() or so, indicating in hte name that it logs

Reply:
I've simply changed this to log_error_errno, which should be enough - one can then enable debug level and try again as per usual